### PR TITLE
Perl version

### PR DESCRIPTION
**Pull request by ghost**
Here's the base of perl version. It may be coded better (idk, it's the first time I take a look at a perl script), but at least it works.

### CODE REVIEW

File: perl.pl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+my $term = $ENV{'TERM'};
+my $shell = $ENV{'SHELL'};
+my $tasks = $#tasks + 1;
+$total = int($total / 1000);
+$avail = int($avail / 1000);
+my $d = int((($times[0] / 60) / 60) / 24);
+my $h = (($times[0] / 60) / 60) % 24;
+my $m = ($times[0] / 60) % 60;
+print $user . "@" . $hostname[0]
+	. "kernel\t" . $version[2]
```
Comment (aeosynth) [inline perl.pl]: even array access is forbidden here by the spec. do it earlier

File: perl.pl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+my @hostname = ffetch("/etc/hostname") ;
+my @version = split / /, (ffetch("/proc/version"))[0];
+my @times = split / /, (ffetch("/proc/uptime"))[0];
+my @fi = ffetch("/proc/meminfo");
+
+( my $total ) = ( $fi[0] =~ /(\d+)/ );
+( my $avail ) = ( $fi[2] =~ /(\d+)/ );
+
+opendir my($dh), "/proc" or die $!;
+my @tasks = grep { /^[+-]?\d+$/ } readdir $dh;
```
Comment (aeosynth) [inline perl.pl]: /proc doesn't use signed integers, so `[+-]` is not needed

File: perl.pl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+closedir $dh;
+
+my $user = $ENV{'USER'};
+my $term = $ENV{'TERM'};
+my $shell = $ENV{'SHELL'};
+my $tasks = $#tasks + 1;
```
Comment (aeosynth) [inline perl.pl]: i'm not sure if this is correct, but we don't do it in the other files, so remove it for regularized sloc

File: perl.pl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	return @r;
+}
+
+my @hostname = ffetch("/etc/hostname") ;
+my @version = split / /, (ffetch("/proc/version"))[0];
+my @times = split / /, (ffetch("/proc/uptime"))[0];
+my @fi = ffetch("/proc/meminfo");
+
+( my $total ) = ( $fi[0] =~ /(\d+)/ );
+( my $avail ) = ( $fi[2] =~ /(\d+)/ );
```
Comment (aeosynth) [inline perl.pl]: can you do this as a (one-liner) map?

**Review by aeosynth — CHANGES_REQUESTED** [review]

File: perl.pl [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+opendir my($dh), "/proc" or die $!;
+my @tasks = grep { /^[+-]?\d+$/ } readdir $dh;
+closedir $dh;
+
+my $user = $ENV{'USER'};
+my $term = $ENV{'TERM'};
+my $shell = $ENV{'SHELL'};
+my $tasks = $#tasks + 1;
+$total = int($total / 1000);
+$avail = int($avail / 1000);
```
Comment (aeosynth) [inline perl.pl]: can you do this when declaring them above?

**Comment by ghost** [comment]
32 to 26 lines

File: perl.pl [hunk 6]
```diff
@@ -5,29 +5,23 @@ sub ffetch {
...
-my @hostname = ffetch("/etc/hostname") ;
-my @version = split / /, (ffetch("/proc/version"))[0];
+my $hostname = (ffetch("/etc/hostname"))[0];
+my $version = (split / /, (ffetch("/proc/version"))[0])[2];
 my @times = split / /, (ffetch("/proc/uptime"))[0];
 my @fi = ffetch("/proc/meminfo");
 
-( my $total ) = ( $fi[0] =~ /(\d+)/ );
-( my $avail ) = ( $fi[2] =~ /(\d+)/ );
+my ($total, $avail) = (int(( $fi[0] =~ /(\d+)/ )[0] / 1000), int(( $fi[2] =~ /(\d+)/ )[0] / 1000));
```
Comment (aeosynth) [inline perl.pl:13]: that's not actually a map

File: perl.pl [hunk 7]
```diff
@@ -5,29 +5,23 @@ sub ffetch {
...
+my $tasks = scalar (grep { /\d+$/ } readdir $dh);
 closedir $dh;
 
-my $user = $ENV{'USER'};
-my $term = $ENV{'TERM'};
-my $shell = $ENV{'SHELL'};
-my $tasks = $#tasks + 1;
-$total = int($total / 1000);
-$avail = int($avail / 1000);
+my ($user, $term, $shell) = ($ENV{'USER'}, $ENV{'TERM'}, $ENV{'SHELL'});
```
Comment (aeosynth) [inline perl.pl]: same

**Pull request merged**